prow/plugins/setmilestone/setmilestone.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package setmilestone implements the `/milestone` command which allows members of the milestone
+// maintainers team to specify a milestone to be applied to an Issue or PR.
+package setmilestone
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "milestone"
+
+var (
+	milestoneRegex   = regexp.MustCompile(`(?m)^/milestone\s+(.+)$`)
+	mustBeSigLead    = "You must be a member of the [kubernetes/kubernetes-milestone-maintainers](https://github.com/orgs/kubernetes/teams/kubernetes-milestone-maintainers/members) github team to set the milestone."
+	invalidMilestone = "The provided milestone is not valid for this repository.  Here are the available milestones:\n %s"
+)
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	SetMilestone(org, repo string, issueNum, milestoneNum int) error
+	ListTeamMembers(id int) ([]github.TeamMember, error)
+	ListMilestones(org, repo string) ([]github.Milestone, error)
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The milestone plugin allows members of the milestone maintainers Github team to set the milestone to an issue or pull request pull request.",
+		Config: map[string]string{
+			"": fmt.Sprintf("The milestone maintainers team is the Github team with ID: %d.", config.MilestoneStatus.MaintainersID),
+		},
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/milestone <milestone number>",
+		Description: "Updates the milestone for an issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Members of the milestone maintainers Github team can use the '/milestone' command. This team is specified in the config by providing the Github team's ID.",
```

Comment:
Second sentence here is not useful to users figuring out what to do, aimed at admins instead. First sentence seems sufficient. Also maybe a GitHub link to the team here would be good?